pkg/trigger/controller.go

```diff
@@ -47,25 +50,40 @@ const (
 )
 
 type MigrationTrigger struct {
```

Comment:
I think we can split the Trigger to two controllers. The storageVersion control loop and the migration control loop is decoupled.